### PR TITLE
Migrate Translator API to v3

### PR DESCRIPTION
**Pull request by enzocanoo**
- Change `Translator.cs` code to use new Translator API v3
- Update unit test and mocks related to the service.
- Update integration tests related to the service.

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Ai.Translation/Model/Alignment.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using Newtonsoft.Json;
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Ai.Translation/Model/Alignment.cs]: Copyright hreader.

File: libraries/Microsoft.Bot.Builder.Ai.Translation/Model/DetectedLanguageModel.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using Newtonsoft.Json;
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Ai.Translation/Model/DetectedLanguageModel.cs]: // (C) Header

Comment (cleemullins) [reply libraries/Microsoft.Bot.Builder.Ai.Translation/Model/DetectedLanguageModel.cs]: Same comment on all these files...

File: libraries/Microsoft.Bot.Builder.Ai.Translation/Model/SentencesLenght.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Ai.Translation.Model
+{
+    public class SentencesLenght
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Ai.Translation/Model/SentencesLenght.cs]: Spelling. SentencesLength

**Comment by enzocanoo** [comment]
@cleemullins, we rebased this PR on top of the `master` branch and dropped the commit related to the integration tests.

**Comment by cleemullins** [comment]
This PR looks good to me. I would like to see @amr-elsehemy  or @chrimc62 review as well for sanity, as they are the code owners. 

Once one of them signs off (or someone they suggest) we can merge. We'll then have to integrate the same set of changes into the Node codebase.

File: libraries/Microsoft.Bot.Builder.Ai.Translation/Model/DetectedLanguageModel.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Ai.Translation.Model
+{
+    public class DetectedLanguageModel
+    {
+        [JsonProperty("language")]
```
Comment (amr-elsehemy) [inline libraries/Microsoft.Bot.Builder.Ai.Translation/Model/DetectedLanguageModel.cs]: I suggest adding some documentation for the new classes to become easier to understand.

File: libraries/Microsoft.Bot.Builder.Ai.Translation/Translator.cs [hunk 5]
```diff
@@ -136,71 +110,52 @@ public Translator(string apiKey, HttpClient httpClient = null)
             }
 
             // body of http request
```
Comment (amr-elsehemy) [inline libraries/Microsoft.Bot.Builder.Ai.Translation/Translator.cs]: I don't think this comment is correct now, it was related to the old variable 'body'

**Comment by enzocanoo** [comment]
Thanks for your feedback @amr-elsehemy. The added models are not intended to be used outside the translator so we convert them to `internal`. However, we also added the documentation from v3 Translator Text API reference ([link](https://docs.microsoft.com/en-us/azure/cognitive-services/translator/reference/v3-0-reference)).

File: libraries/Microsoft.Bot.Builder.Ai.Translation/Model/Alignment.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Ai.Translation.Model
+{
+    public class Alignment
+    {
+        [JsonProperty("proj")]
```
[the remark this answers is not in the document]
Comment (enzocanoo) [reply libraries/Microsoft.Bot.Builder.Ai.Translation/Model/Alignment.cs]: The property reflect the json response from Translator API v3 ([link](https://docs.microsoft.com/en-us/azure/cognitive-services/translator/reference/v3-0-translate?tabs=curl#response-body)).

**Pull request merged**